### PR TITLE
Adding tap duration to Tapping Task Issue #561

### PR DESCRIPTION
**Pull request by m1entus**
This PR adds a duration to ORKTappingSample.

When user press the button and don't release finger before times up, then i measure `duration` when step is finished for last sample.

### CODE REVIEW

**Comment by YuanZhu-apple** [comment]
@umerkhan-apple Can you look at this one?

**Comment by syoung-smallwisdom** [comment]
@m1entus 
Hi Michael, We are also using the tapping test in our applications. In our case, we still support the iPhone 4S and I am concerned that the CPU has a risk of getting bogged down by this implementation.

Any reason you don't just use pointers to hold the value of the last sample for each button?

Example:
````
@interface ORKTappingIntervalStepViewController () <UIGestureRecognizerDelegate>

@property (nonatomic, strong) NSMutableArray *samples;
@property (nonatomic, strong) NSMutableDictionary <NSNumber *, ORKTappingSample *> *lastSamples;

@end

/// ...

- (IBAction)buttonPressed:(id)button forEvent:(UIEvent *)event {
    
    if (self.samples == nil) {
        // Start timer on first touch event on button
        _samples = [NSMutableArray array];
        _lastSamples = [NSMutableDictionary dictionary];
        _hitButtonCount = 0;
        [self start];
    }

/// ...

    _lastSamples[@(buttonIdentifier)] = sample;
    [self.samples addObject:sample];

/// ...

    // Take last sample for buttonIdentifier, and fill duration
    ORKTappingSample *sample = _lastSample[buttonIdentifier];
    sample.duration = mediaTime - sample.timestamp - _tappingStart;

````

File: ResearchKit/Common/ORKResult.m [hunk 1]
```diff
@@ -180,20 +182,22 @@ - (BOOL)isEqual:(id)object {
     __typeof(self) castObject = object;
     
     return ((self.timestamp == castObject.timestamp) &&
+            (self.duration == castObject.duration) &&
```
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKResult.m]: also need to override `hash`

**Comment by umerkhan-apple** [comment]
r=m3! 👍 
@YuanZhu-apple Friendly ping before I merge.

File: ResearchKit/ActiveTasks/ORKTappingIntervalStepViewController.m [hunk 2]
```diff
@@ -197,6 +243,12 @@ - (IBAction)buttonPressed:(id)button forEvent:(UIEvent *)event {
     [self receiveTouch:[[event touchesForView:button] anyObject] onButton:index];
 }
 
+- (IBAction)buttonReleased:(id)button forEvent:(UIEvent *)event {
+    NSInteger index = (button == _tappingContentView.tapButton1) ? ORKTappingButtonIdentifierLeft : ORKTappingButtonIdentifierRight;
```
Comment (YuanZhu-apple) [inline ResearchKit/ActiveTasks/ORKTappingIntervalStepViewController.m]: @m1entus Can you make this small change? Thx!

Comment (m1entus) [inline ResearchKit/ActiveTasks/ORKTappingIntervalStepViewController.m]: Sure will do it today

**Comment by m1entus** [comment]
@YuanZhu-apple done.

**Pull request merged**